### PR TITLE
WIP: Add Posix_Waiter APIs for SB16 and tests

### PR DESCRIPTION
**Pull request by maxz-lab**
b/302741384

### CODE REVIEW

File: starboard/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,11 @@ since the version previous to it.
 
 ## Version 16
 
+### Deprecated SbSocketWaiter APIs and migrated to POSIX based APIs
+Introduce SB16 specific APIs that takes integer as socket descriptor.
```
Comment (y4vor) [inline starboard/CHANGELOG.md:13]: Remove SB16. This whole section is about Version 16.

File: starboard/elf_loader/exported_symbols.cc [hunk 2]
```diff
@@ -477,6 +477,8 @@ ExportedSymbols::ExportedSymbols() {
   REGISTER_SYMBOL(munmap);
   REGISTER_SYMBOL(open);
   REGISTER_SYMBOL(posix_memalign);
+  REGISTER_SYMBOL(PosixSocketWaiterAdd);
```
Comment (y4vor) [inline starboard/elf_loader/exported_symbols.cc:480]: Guard with SB16?

File: starboard/tools/api_leak_detector/api_leak_detector.py [hunk 3]
```diff
@@ -109,6 +109,8 @@
     'lseek',
     'malloc',
     'posix_memalign',
+    'PosixSocketWaiterAdd',
```
Comment (y4vor) [inline starboard/tools/api_leak_detector/api_leak_detector.py:112]: why do you need these?

File: starboard/socket_waiter.h [hunk 4]
```diff
@@ -79,6 +79,14 @@ typedef void (*SbSocketWaiterCallback)(SbSocketWaiter waiter,
                                        void* context,
                                        int ready_interests);
 
+#if SB_API_VERSION >= 16
+// Function pointer for socket waiter callbacks.
+typedef void (*PosixSocketWaiterCallback)(SbSocketWaiter waiter,
```
Comment (y4vor) [inline starboard/socket_waiter.h:84]: Those need to be `SbPosixSocketWaiterCallback` etc... 

All the SB APIs start with `Sb`.

File: starboard/tools/api_leak_detector/api_leak_detector.py (hunk 3 above)
Comment (maxz-lab) [reply starboard/tools/api_leak_detector/api_leak_detector.py]: They are renamed to SbPosixSocketWaiter* and moved to near other SbSocketWaiter functions in this file.

File: starboard/elf_loader/exported_symbols.cc (hunk 2 above)
Comment (maxz-lab) [reply starboard/elf_loader/exported_symbols.cc]: Guarded by SB16.

File: starboard/tools/api_leak_detector/api_leak_detector.py [hunk 5]
```diff
@@ -109,6 +109,8 @@
     'lseek',
     'malloc',
     'posix_memalign',
+    'SbPosixSocketWaiterAdd',
```
Comment (y4vor) [inline starboard/tools/api_leak_detector/api_leak_detector.py:112]: We don't need those, right?

Comment (maxz-lab) [reply starboard/tools/api_leak_detector/api_leak_detector.py]: Right, they are not needed.

**Pull request closed without merging**